### PR TITLE
Cleanup repository

### PR DESCRIPTION
**Pull request by milang**
- Switch the whole repository to UNIX-only newlines
- Switch to yarn 3.2.3
- Upgrade NPM dependencies
- Add a script that removes `devDependencies`, `scripts`, and `eslinkConfig` from `package.json` before packing
- Add publishing instructions to `README.md`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -125,50 +153,56 @@ The following example will show you how to:
...
-## Testing
+## Contributing
+
+### Testing
 
 To test you can install the npm dependencies:
 
-    npm install
+```sh
+npm install
```
Comment (101100) [inline README.md:177]: Should this be `yarn`?

File: README.md [hunk 2]
```diff
@@ -125,50 +153,56 @@ The following example will show you how to:
...
+```
 
 and then use:
 
-    jake test
+```sh
+jake test
```
Comment (101100) [inline README.md:183]: This should be `yarn test`

File: README.md [hunk 3]
```diff
@@ -125,50 +153,56 @@ The following example will show you how to:
...
 
-## Documentation
+### Documentation
 
-Documentation is build with Leafdoc, to generate the documentation use
+Documentation is built with Leafdoc, to generate the documentation use
 
-    jake docs
+```sh
+jake docs
```
Comment (101100) [inline README.md:191]: Here too.

File: build/clean-package-json.mjs [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import { readFileSync, writeFileSync } from 'node:fs';
```
Comment (101100) [inline build/clean-package-json.mjs:4]: Maybe add a comment:

```ts
// Script run during packaging (see package.json) to tidy up the package.json
// that is included with the released version.
```

**Review by michaelfarrelly — APPROVED** [review]
Changes look good. 

Verified commands ran locally: `yarn` (uses yarn3 - very nice), `yarn build`, `yarn test`, `yarn prepack` and `yarn pack` all successful without error.

**Review by 101100 — CHANGES_REQUESTED** [review]
Done with round 1.

File: README.md (hunk 1 above)
Comment (milang) [reply README.md:177]: I was not rewriting this part of the README, just fixing Markdown Validator issues. We have a "Pandell section" at the top of the README, this is part of the original instructions.

File: README.md (hunk 2 above)
Comment (milang) [reply README.md:183]: Ditto for just applying Markdown Validator fixes, not changing content of the original.

File: README.md (hunk 3 above)
Comment (milang) [reply README.md:191]: Ditto for just applying Markdown Validator fixes, not changing content of the original.

File: build/clean-package-json.mjs (hunk 4 above)
Comment (milang) [reply build/clean-package-json.mjs:4]: Great idea, added 👍🏼.

**Comment by milang** [comment]
@101100 addressed feedback, ready for next round.

File: README.md (hunk 1 above)
Comment (101100) [reply README.md:177]: Sounds good

**Review by 101100 — APPROVED** [review]
Looks good to me. 🍪

**Pull request merged**